src/sql/workbench/services/notebook/browser/models/modelInterfaces.ts

```diff
@@ -548,6 +549,15 @@ export interface ICellModel {
 	 * Returns the name of the attachment added to metadata.
 	 */
 	addAttachment(mimeType: string, base64Encoding: string, name: string): string;
+	richTextCursorPosition: ICaretPosition;
+	markdownCursorPosition: IPosition;
+}
+
+export interface ICaretPosition {
```

Comment:
Why do you need to create a new type here instead of just using an existing one such as IPosition?

Reply:
IPositon only has linenumber and column and it's defined for code editors context, here I need to track the caret position on html nodes so I created a new type instead overriding that.